### PR TITLE
Make e2e test results available in Bitrise

### PR DESCRIPTION
**Pull request by NicolasMassart**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

## Description

Update existing bitrise upload steps and make the Android test results always uploaded.
Also add reporter to generate test on iOS e2e and upload.

1. What is the reason for the change?
    - Test were not pushed to Bitrise test tab on failure (and not even on success by the way)

2. What is the improvement/solution?
    - update upload to Bitrise step version as v1 is not compatible anymore ([leads to not uploading anything](https://github.com/MetaMask/mobile-planning/issues/733#issuecomment-1497783953))
    - update android wdio workflow to always publish test reports, even on failure
    - add custom reporter in Detox config
    - add [Jest Junit reporter](https://socket.dev/npm/package/jest-junit/overview/15.0.0) as a dev dependency for the custom Detox reporter
    - add `custom-test-results-export@1` steps in iOS e2e workflow to search for reports and upload them. The default copy script was not working.

## Issue

fixes MetaMask/mobile-planning#733

### Android e2e (Appium/wdio)
- One Android test ran on my own Bitrise setup but given the amount time required to run tests, I haven't ran them again on Metamask Bitrise. But it works. Feel free to run a test if you want, they fail for now, so this would require to run only a simple one that passes (making Android e2e pass is clearly not part of this PR).
   image

### iOS e2e (Detox)
iOS e2e are not passing because of some deeplink issues, so I ran them with only individual test because I only needed to check the report part:
- successful basic test: available at https://app.bitrise.io/build/807e2429-77ac-494a-ac88-d2b8abc75363?tab=tests
  image

- failing basic test: https://app.bitrise.io/build/169fb2a2-134b-44a1-a04d-b2741ecf6bf8?tab=tests
  image

## Checklist

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

**Review by sethkfman — APPROVED** [review]
LGTM

**Review by cortisiko — APPROVED** [review]
ty @NicolasMassart !! 🌮 🌮 🌮

**Comment by cortisiko** [comment]
alright @NicolasMassart I was _unable_ to generate the test report on a failed test build: https://app.bitrise.io/build/8437ed12-96ac-4303-aa1f-283ef9cb9b03

I get this: 

> Collecting test results...
> No test results found


You can test this the quick and easy way by using the following env variables such as `BROWSERSTACK_TAG_EXPRESSION` and `BROWSERSTACK_APP_URL`. You can read up more [here](https://www.notion.so/Mobile-E2E-Testing-361fee8ff65048f0b18d6d3dddc5ad9c?pvs=4#2bfdb6b911f4484f9efd1d5e74ab26c8)


Ideally when you kick off the e2e build, you want to pass in the browser stack app url  as well as the tag you want to run. The tag with one 1 test is `Performance` so you should probably use that one. 

So to be clear to generate the test results quickly on a test build:

- in bitrise go to the advanced section when you are about to kick off a build
- scroll to the Custom Environment Variables section
- set the BROWSERSTACK_APP_URL env var to: bs://b8f0203f25deef45e24fa288a10592c6a4daa7b3
- the BROWSERSTACK_TAG_EXPRESSION to: Performance

then the work flow you should use is: `wdio_android_e2e_test`

File: package.json [hunk 1]
```diff
@@ -152,9 +152,9 @@
     "@metamask/message-manager": "^1.0.1",
     "@metamask/network-controller": "^2.0.0",
     "@metamask/permission-controller": "^1.0.2",
+    "@metamask/phishing-controller": "^2.0.0",
```
Comment (Fatxx) [inline package.json:155]: Do we need this dependency?

Comment (NicolasMassart) [reply package.json:155]: It was just reorganised in the list, was already there.

**Pull request merged**